tests/unit/synapseutils/unit_test_synapseutils_sync.py

```diff
@@ -351,6 +351,53 @@ def test_extract_file_entity_metadata__ensure_correct_row_metadata(syn):
                 assert file_entity.get(key) == file_row_data.get(key)
 
 
+def test_manifest_upload(syn):
+    """Verify behavior of synapseutils.sync._manifest_upload"""
+
+    data = {
+        'path': ['/tmp/foo', '/tmp/bar', '/tmp/baz'],
+        'parent': ['syn123', 'syn456', 'syn789'],
+        'name': ['foo', 'bar', 'baz'],
+        'used': [None, '/tmp/foo', '/tmp/bar'],
+        'executed': [None, None, '/tmp/foo'],
+        'anno_1': ['', 'v1', 'v2'],
+        'anno_2': ['v3', 'v4', ''],
+    }
+
+    df = pd.DataFrame(data=data)
+
+    with patch.object(synapseutils.sync, '_SyncUploadItem') as upload_item_init, \
+            patch.object(synapseutils.sync, '_SyncUploader') as uploader_init:
+        synapseutils.sync._manifest_upload(syn, df)
+
+    upload_items = []
+    for i in range(3):
+        expected_path = data['path'][i]
+        expected_parent = data['parent'][i]
+        expected_name = data['name'][i]
+        expected_used = data['used'][i]
+        expected_executed = data['executed'][i]
+
+        expected_annos = {}
+        expected_annos.update({'anno_1': data['anno_1'][i]})
+        expected_annos.update({'anno_2': data['anno_2'][i]})
+
+        # empty strings values should not be uploaded as annotations
+        expected_annos = {k: v for k, v in expected_annos.items() if v != ''}
```

Comment:
This test generates expected annotations in the same way as the code being tested. Prefer an explicit dict.

Copy the `data` dict and overwrite keys/value pairs that should have changed